### DISCUSSION TITLE
Missing argument in createContext when called from getServerSideProps

### DISCUSSION THREAD

**Discussion by Brenndoerfer** [trpc/trpc#3559] [General]
I've looked at the documentation for the SSG Helper but it seems to be missing something (https://trpc.io/docs/ssg-helpers)

When constructing the `ssg` object as below, `createContext()` usually expects `opts` as an input. 
```ts
 const ssg = createProxySSGHelpers({
    router: appRouter,
    ctx: await createContext(),
    transformer: superjson,
  });
```

Corresponding `createContext` 

```ts
export const createContext = async (opts: CreateNextContextOptions) => {
  let session = null;
  if (opts?.req && opts?.res) {
    const { req, res } = opts;
    session = await getTokenFromHeader(req);
  }
  return await createContextInner({
    session,
  });
};
```

The code works, but I am getting a ts error: 
```ts
An argument for 'opts' was not provided
```

I could get around it by casting:
```ts
ctx: await createContext({} as CreateNextContextOptions)
```

but i was wondering if there is a more elegant/ correct solution?

Thanks

**Comment by KATT** [comment]
Call `createContextInner` instead

**Reply by KATT** [reply]
https://trpc.io/docs/context#inner-and-outer-context

**Reply by Brenndoerfer** [reply]
Thanks, that worked!
